features/step_definitions/comment_steps.rb

```diff
@@ -302,3 +302,8 @@
     click_link("Yes, delete!") # TODO: Fix along with comment deletion.
   end
 end
+
+When /^I mark the comment as spam$/ do
+  click_link("Spam")
+  expect(page.accept_alert).to eq("Are you sure you want to mark this as spam?") if @javascript
```

Comment:
Could you break this out into a separate step, like `When "I confirm I want to mark the comment as spam" do`? 

With the confirmation in the same `When` as pressing the spam button, the difference in JavaScript/non-JavaScript behavior isn't super clear when reading the two scenarios (i.e., the JavaScript test says we expect a confirmation popup, but it's not obvious what step is dealing with that).